Xbim.Common/Configuration/XbimServices.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Xbim.Essentials.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010029a3c6da60efcb3ebe48c3ce14a169b5fa08ffbf5f276392ffb2006a9a2d596f5929cf0e68568d14ac7cbe334440ca0b182be7fa6896d2a73036f24bca081b2427a8dec5689a97f3d62547acd5d471ee9f379540f338bbb0ae6a165b44b1ae34405624baa4388404bce6d3e30de128cec379147af363ce9c5845f4f92d405ed0")]
+
+namespace Xbim.Common.Configuration
+{
+    /// <summary>
+    /// Class encapsulating the services used in the application managed by Dependency Injection
+    /// </summary>
+    public class XbimServices
+    {
+
+        private XbimServices()
+        {
+            Rebuild();
+        }
+
+        /// <summary>
+        /// For testing only
+        /// </summary>
+        /// <returns></returns>
+        internal static XbimServices CreateInstance()
+        {
+            return new XbimServices();
+        }
+
+        private bool isBuilt = false;
+
+        private IServiceCollection services = new ServiceCollection();
+
+        static Lazy<XbimServices> lazySingleton = new Lazy<XbimServices>(() => new XbimServices());
+        Lazy<IServiceProvider> serviceProviderBuilder;
+
+        /// <summary>
+        /// The shared instance of all Xbim Services
+        /// </summary>
+        public static XbimServices Current { get; private set; } = lazySingleton.Value;
```

Comment:
Singleton service for all consumers. It needs to be discoverable somewhere

CreateInstance is just for testing an isolated instance